### DISCUSSION TITLE
Isolating device to boot device.

### DISCUSSION THREAD

**Discussion by pendrivelinux** [a1ive/grub2-filemanager#311] [Q&A] [answered]
Hey a1live,

Is it possible to isolate menu navigation to only the device the filemanager was booted from? 

Regardless of where I set a condition to do this, it always seems possible to use esc to go back up the menu and select other drives and devices.

**Comment by a1ive** [comment]
Not supported.
This is because the file manager does not know which device it is booting from.

**Reply by pendrivelinux** [reply]
I can set the device. I am just unsure of where I should force the file manager to only use the device I set.

if test "${device}" = "${setdev}"; then
what to wrap to only enumerate this device?
fi;

**Comment by a1ive** [comment, marked as answer]
https://github.com/a1ive/grub/commit/68ef91230011724e2e3cbc73238a1860c4d9f4c0
Now you could use `grubfm_set --top` to set top-level dirs.
Assuming you only want to enumerate files under (hd0,msdos1):
Create file `/boot/grubfm/config` under `(hd0,msdos1)` with the following content
```
configfile (${user})/boot/grubfm/config.cfg
```
Create file `/boot/grubfm/config.cfg` under `(hd0,msdos1)` with the following content
```
grubfm_set --top "(${user})/"
grubfm "(${user})/"
```
![深度截图_选择区域_20201230214111](https://user-images.githubusercontent.com/10670106/103354916-cbadd500-4ae7-11eb-9e3f-c19afb37b538.png)

**Reply by pendrivelinux** [reply]
Hi a1ive,

I just tried with the most recent build, and your solution appears to work. 
Thank you!
